internal/api/core/route.go

```diff
@@ -0,0 +1,117 @@
+package core
+
+import (
+	"net/http"
+	"reflect"
+	"regexp"
+
+	"github.com/edulinq/autograder/internal/log"
+)
+
+type Route interface {
+	GetMethod() string
+	GetRegex() *regexp.Regexp
+	GetBasePath() string
+	Handle(response http.ResponseWriter, request *http.Request) error
+}
+
+// Inspired by https://benhoyt.com/writings/go-routing/
+type BaseRoute struct {
+	Method   string
+	BasePath string
+	Regex    *regexp.Regexp
+	Handler  RouteHandler
+}
+
+type APIRoute struct {
+	BaseRoute
+	Description string
+	Request     reflect.Type
+	Response    reflect.Type
```

Comment:
These fields should be suffixed with "Type" (since they are types and not the values themselves).